cylc/flow/scheduler.py

```diff
@@ -751,7 +751,14 @@ def restart_remote_init(self):
         """Remote init for all submitted/running tasks in the pool."""
         self.task_job_mgr.task_remote_mgr.is_restart = True
         distinct_install_target_platforms = []
+
         for itask in self.pool.get_tasks():
+            # get a platform if the task proxy hasn't got one:
+            if not itask.platform:
+                itask.platform = get_platform(
+                    itask.tdef.rtconfig, bad_hosts=self.bad_hosts
+                )
```

Comment:
I think that this is a possible bug.

If a running task does *not* have a platform set, we cannot safely guess it from the configuration because the configuration may have changed during restart. Also if the platform is a `platform-group` we may assign the wrong platform to the task. Either case would result in a polling failure.

It's ok for tasks not to have a platform set on restart provided they are in the waiting state (because we wouldn't expect the platform to be set at that point).

Reply:
If you don't have it the next line down will break:
```
itask.platform['install target'] = (
                get_install_target_from_platform(itask.platform))
 ```

Reply:
If a task doesn't have a platform, that means it hasn't been run yet, declaring a platform would be incorrect.

So I think this should be:

```
if not itask.platform:
    continue
```